x/incentive/keeper/payout_test.go

```diff
@@ -99,178 +100,260 @@ func (suite *KeeperTestSuite) setupExpiredClaims() {
-				vesting.Period{Length: int64(5), Amount: cs(c("ukava", 100))},
+		{
+			name: "add coins to existing period",
+			args: args{
+				accArgs: accountArgs{
+					periods: vesting.Periods{
+						vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
+						vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
+						vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
+						vesting.Period{Length: 5, Amount: cs(c("ukava", 5))}},
+					origVestingCoins: cs(c("ukava", 20)),
+					startTime:        100,
+					endTime:          120,
+				},
+				period:              vesting.Period{Length: 5, Amount: cs(c("ukava", 6))},
+				ctxTime:             time.Unix(110, 0),
+				mintModAccountCoins: true,
+				expectedPeriods: vesting.Periods{
+					vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
+					vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
+					vesting.Period{Length: 5, Amount: cs(c("ukava", 11))},
+					vesting.Period{Length: 5, Amount: cs(c("ukava", 5))}},
+				expectedStartTime: 100,
+				expectedEndTime:   120,
+			},
+			errArgs: errArgs{
+				expectErr: false,
+				contains:  "",
 			},
-			expectedOriginalVesting: cs(c("ukava", 1000)),
-			expectedCoins:           cs(c("ukava", 1000)),
-			expectedStartTime:       int64(90),
-			expectedEndTime:         int64(130),
 		},
-		vestingAccountTest{
-			name:                    "insufficient module account balance",
-			blockTime:               time.Unix(90, 0),
-			args:                    args{coins: cs(c("ukava", 1000)), length: 11},
-			errArgs:                 errArgs{expectErr: true, errType: types.ErrInsufficientModAccountBalance},
-			expectedPeriods:         vesting.Periods{},
-			expectedOriginalVesting: sdk.Coins{},
-			expectedCoins:           sdk.Coins{},
-			expectedStartTime:       int64(0),
-			expectedEndTime:         int64(0),
+		{
+			name: "insufficient mod account balance",
+			args: args{
+				accArgs: accountArgs{
+					periods: vesting.Periods{
+						vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
+						vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
+						vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
+						vesting.Period{Length: 5, Amount: cs(c("ukava", 5))}},
+					origVestingCoins: cs(c("ukava", 20)),
+					startTime:        100,
+					endTime:          120,
+				},
+				period:              vesting.Period{Length: 7, Amount: cs(c("ukava", 6))},
+				ctxTime:             time.Unix(125, 0),
+				mintModAccountCoins: false,
+				expectedPeriods: vesting.Periods{
+					vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
+					vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
+					vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
+					vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
+					vesting.Period{Length: 12, Amount: cs(c("ukava", 6))}},
+				expectedStartTime: 100,
+				expectedEndTime:   132,
+			},
+			errArgs: errArgs{
+				expectErr: true,
+				contains:  "insufficient funds",
+			},
 		},
```

Comment:
nit: I had a look at the code coverage, and there was a small section uncovered (ln 182 to 188), so this test case hits it.
```suggestion
		},
		{
			name: "add large period mid schedule",
			args: args{
				accArgs: accountArgs{
					periods: vesting.Periods{
						vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
						vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
						vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
						vesting.Period{Length: 5, Amount: cs(c("ukava", 5))}},
					origVestingCoins: cs(c("ukava", 20)),
					startTime:        100,
					endTime:          120,
				},
				period:              vesting.Period{Length: 50, Amount: cs(c("ukava", 6))},
				ctxTime:             time.Unix(110, 0),
				mintModAccountCoins: true,
				expectedPeriods: vesting.Periods{
					vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
					vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
					vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
					vesting.Period{Length: 5, Amount: cs(c("ukava", 5))},
					vesting.Period{Length: 40, Amount: cs(c("ukava", 6))}},
				expectedStartTime: 100,
				expectedEndTime:   160,
			},
			errArgs: errArgs{
				expectErr: false,
				contains:  "",
			},
		},
```